core/vm/privacy/bulletproof.go

```diff
@@ -807,24 +809,42 @@ func (ipp *InnerProdArg) Deserialize(proof []byte, numChallenges int) error {
 	if err != nil {
 		return err
 	}
+	for i := 0; i < len(L); i++ {
+		if !curve.(*secp256k1.BitCurve).IsOnCurve(L[i].X, L[i].Y) || L[i].isZeroPoint() {
+			return errors.New("invalid ECPoint, point is (0,0) or not on curve")
+		}
+	}
 	ipp.L = append(ipp.L, L[:]...)
 	offset += len(L) * 33
 
 	R, err := deserializePointArray(proof[offset:], uint32(numChallenges)-1)
 	if err != nil {
 		return err
 	}
-
+	for i := 0; i < len(R); i++ {
+		if !curve.(*secp256k1.BitCurve).IsOnCurve(R[i].X, R[i].Y) || R[i].isZeroPoint() {
+			return errors.New("invalid ECPoint, point is (0,0) or not on curve")
+		}
+	}
 	ipp.R = append(ipp.R, R[:]...)
 	offset += len(R) * 33
 
 	if len(proof) <= offset+64+4 {
 		return errors.New("proof data too short")
 	}
 
-	ipp.A = new(big.Int).SetBytes(proof[offset : offset+32])
+	A := new(big.Int).SetBytes(proof[offset : offset+32])
+	if A.Cmp(big.NewInt(0)) == 0 || A.CmpAbs(curve.Params().N) != 1 {
+		return errors.New("invalid scalar, scalar should be nonzero within the range of the scalar field")
+	}
+	ipp.A = A
 	offset += 32
-	ipp.B = new(big.Int).SetBytes(proof[offset : offset+32])
+
+	B := new(big.Int).SetBytes(proof[offset : offset+32])
+	if B.Cmp(big.NewInt(0)) == 0 || B.CmpAbs(curve.Params().N) != 1 {
+		return errors.New("invalid scalar, scalar should be nonzero within the range of the scalar field")
```

Comment:
This error `errors.New("invalid scalar, scalar should be nonzero within the range of the scalar field")` occurs 8 times.